File: concept/17_recursion/says_digit/says_digit.cpp


#include <iostream>
using namespace std;

int saysDigit(int n)
{
    string arr[10] = {
        "zero",
        "one",
        "two",
        "three",
        "four",
        "five",
        "six",
        "seven",
        "eight",
        "nine",
    };
    if (n % 10 == 0)
    {
        return n;
    }

    int digit = n % 10;
    n = n / 10;

    saysDigit(n);
    cout << arr[digit] << " ";

    return digit;
}

int main()
{

    cout << "Enter n : ";
    int n;
    cin >> n;

    saysDigit(n);
}

File: concept/17_recursion/says_digit/says_digit.dart
import 'dart:io';

void main() {
  print("Enter number : ");
  int n = int.parse(stdin.readLineSync()!);
  sayDigits(n);
}

void sayDigits(int n) {
  List<String> arr = [
    "zero",
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
  ];

  if (n % 10 == 0) {
    return;
  }

  int digit = n % 10;
  n = n ~/ 10;

  sayDigits(n);

  stdout.write(arr[digit]);
  stdout.write(" ");
  return;
}
